packages/mdl-snackbar/README.md

```diff
@@ -0,0 +1,145 @@
+| multiline | Whether to show the snackbar with space for multiple lines of text | Optional |  Boolean |
+| actionOnBottom | Whether to show the action below the multiple lines of text | Optional, applies when multiline is true |  Boolean |
+
+### Using the Foundation Class
+
+MDL Snackbar ships with an `MDLSnackbarFoundation` class that external frameworks and libraries can
+use to integrate the component. As with all foundation classes, an adapter object must be provided.
+The adapter for snackbars must provide the following functions, with correct signatures:
+
+| Method Signature | Description |
+| --- | --- |
+| `addClass(className: string) => void` | Adds a class to the root element. |
+| `removeClass(className: string) => void` | Removes a class from the root element. |
+| `setAriaHidden() => void` | Sets `aria-hidden="true"` on the root element. |
+| `unsetAriaHidden() => void` | Removes the `aria-hidden` attribute from the root element. |
+| `setMessageText(message: string) => void` | Set the text content of the message element. |
+| `setActionText(actionText: string) => void` | Set the text content of the action element. |
+| `setActionAriaHidden() => void` | Sets `aria-hidden="true"` on the action element. |
+| `unsetActionAriaHidden() => void` | Removes the `aria-hidden` attribute from the action element. |
+| `registerActionClickHandler(handler: EventListener) => void` | Registers an event handler to be called when a `click` event is triggered on the action element. |
+| `deregisterActionClickHandler(handler: EventListener) => void` | Deregisters an event handler from a `click` event on the action element. This will only be called with handlers that have previously been passed to `registerActionClickHandler` calls. |
+| `registerTransitionEndHandler(handler: EventListener) => void` | Registers an event handler to be called when an `transitionend` event is triggered on the root element. Note that you must account for
+vendor prefixes in order for this to work correctly. |
+| `deregisterTransitionEndHandler(handler: EventListener) => void` | Deregisters an event handler from an `transitionend` event listener. This will only be called with handlers that have previously been passed to `registerTransitionEndHandler` calls. |
+| `setTimeout(func: Function, delay: Numeric) => Numeric` | Register a function to be called after at least a certain delay. |
+
+## Avoiding Flash-Of-Unstyled-Content (FOUC)
+
+If you are loading the `mdl-snackbar` CSS asynchronously, you may experience a brief flash-of-unstyled-content (FOUC) due to the
+snackbar's translate transition running once the CSS loads. To avoid this temporary FOUC, you can add the following simple style
+before the `mdl-snackbar` CSS is loaded:
+
+```css
+.mdl-snackbar { transform: translateY(100%); }
+```
+This will move the snackbar offscreen until the CSS is fully loaded and avoids a translate transition upon load.
+
+## Theming
```

Comment:
FWIW it [has landed](../../tree/master/packages/mdl-theme), but the snackbar doesn't really need to be themed.